packages/-ember-data/tests/integration/store/store-creation-recursion-test.js

```diff
@@ -11,7 +11,7 @@ module('integration/store/creation-recursion', function(hooks) {
     let storeFactory = this.owner.factoryFor('service:store');
 
     this.owner.unregister('service:store');
-    this.owner.register('service:store', storeFactory);
+    this.owner.register('service:store', storeFactory.class);
```

Comment:
What caused this change?

Reply:
Ember asserts if you pass the factoryFor result directly to register now